emitter.py

```diff
@@ -124,13 +115,57 @@ def http_emitter(message, log, agentConfig, endpoint):
 
     except Exception:
         log.exception("Unable to post payload.")
-        try:
-            log.error("Received status code: {0}".format(r.status_code))
-        except Exception:
-            pass
 
 
-def post_headers(agentConfig, payload):
+def split_payload(legacy_payload):
+    metrics = list(legacy_payload['metrics'])
```

Comment:
Nitpick: the method performs reads only on `metrics`, thus, it is not required to make a copy of it.